packages/serverpod/lib/src/database/extensions.dart

```diff
@@ -101,150 +86,275 @@ extension TableComparisons on TableDefinition {
     return null;
   }
 
-  /// Returns true if the table structure is identical to the [other]. Ignores
-  /// comparisons of Dart types and names.
-  bool like(TableDefinition other) {
-    // Columns
-    if (other.columns.length != columns.length) {
-      return false;
+  /// Compares this table definition with [other], returning a list of mismatches.
+  /// The comparison checks columns, indexes, foreign keys, and general table properties.
+  /// Returns an empty list if the tables are identical.
+  List<String> like(TableDefinition other) {
+    List<String> mismatches = [];
+
+    // Compare table name
+    if (other.name != name) {
+      mismatches
+          .add('Table name mismatch: Expected "$name", found "${other.name}".');
+    }
+
+    // Compare table space
+    if (other.tableSpace != tableSpace) {
+      mismatches.add(
+          'Tablespace mismatch: Expected "$tableSpace", found "${other.tableSpace}".');
     }
+
+    // Compare table schema
+    if (other.schema != schema) {
+      mismatches
+          .add('Schema mismatch: Expected "$schema", found "${other.schema}".');
+    }
+
+    // Columns
     for (var column in columns) {
       var otherColumn = other.findColumnNamed(column.name);
-      if (otherColumn == null || !column.like(otherColumn)) {
-        return false;
+      if (otherColumn == null) {
+        mismatches
+            .add('Column "${column.name}" is missing in the target schema.');
+      } else {
+        var columnMismatches = column.like(otherColumn);
+        if (columnMismatches.isNotEmpty) {
+          mismatches.add(
+              'Column "${column.name}" mismatch: ${columnMismatches.join(', ')}');
+        }
       }
     }
 
-    // Indexes
-    if (other.indexes.length != indexes.length) {
-      return false;
+    for (var otherColumn in other.columns) {
+      var column = findColumnNamed(otherColumn.name);
+      if (column == null) {
+        mismatches.add(
+            'Extra column "${otherColumn.name}" found in the target schema.');
+      }
     }
+
+    // Indexes
     for (var index in indexes) {
-      var otherIndex = other.findIndexNamed(
-        index.indexName,
-        ignoreCase: true,
-      );
-      if (otherIndex == null ||
-          !index.like(
-            otherIndex,
-            ignoreCase: true,
-          )) {
-        return false;
+      var otherIndex = other.findIndexNamed(index.indexName, ignoreCase: true);
+      if (otherIndex == null) {
+        mismatches
+            .add('Index "${index.indexName}" is missing in the target schema.');
+      } else {
+        var indexMismatches = index.like(otherIndex, ignoreCase: true);
+        if (indexMismatches.isNotEmpty) {
+          mismatches.add(
+              'Index "${index.indexName}" mismatch: ${indexMismatches.join(', ')}');
```

Comment:
Suggestion: Improve formatting for sublists.

When these issues stack up the output here will not be so nice. Everything will be bundled on a single line.

Would it be possible to print these as sublists indexed correctly so that it is clear that all these issues relate to an index?

Reply:
Now this is not an issue anymore, fixed it by introducing comparison warning classes.